WebApp/autoreduce_webapp/reduction_viewer/views.py

```diff
@@ -277,20 +275,28 @@ def run_summary(_, instrument_name=None, run_number=None, run_version=0):
                               'reduction_location': reduction_location,
                               'started_by': started_by}
 
-        if reduction_location:
+    except PermissionDenied:
+        raise
+    except Exception as exception:
+        # Error that we cannot recover from - something wrong with instrument, run, or experiment
+        LOGGER.error(exception)
+        return {"message": str(exception)}
+
+    if reduction_location:
+        try:
             plot_handler = PlotHandler(instrument_name=run.instrument.name,
                                        rb_number=rb_number,
                                        run_number=run.run_number,
                                        server_dir=reduction_location)
             plot_locations = plot_handler.get_plot_file()
             if plot_locations:
                 context_dictionary['plot_locations'] = plot_locations
-
-    except PermissionDenied:
-        raise
-    except Exception as exception:
-        LOGGER.error(exception)
-        context_dictionary = {}
+        except Exception as exception:
+            # Lack of plot images is recoverable - we shouldn't stop the whole page rendering
+            # if something is wrong with the plot images - but display an error message
+            err_msg = "Encountered error while retrieving plots for this run"
+            LOGGER.error(f"{err_msg}. Run {run}\n. Error: {exception}")
+            context_dictionary["plot_error_message"] = f"{err_msg}. Error: {str(exception)}"
```

Comment:
Just putting this here for discussion. Should we be displaying technical error messages to users, or is it enough to display a user style error message but also log the technical details separately? In other words would a error relating to the code be useful for a user to see?

Reply:
Good point. I think in this case its fine but is worth looking at in future. I don't know if Django (my lack of Django knowledge) has something similar to custom error pages in .net that can be toggled on or off (on for prod)

Reply:
I agree that printing the exception itself might be a bad idea - if something goes very wrong it could just fill up the whole page. I'll remove it from the user-facing error message

Reply:
I agree too. On top of the UX reason you've mentioned there, I've heard it's bad practice security wise to let a user know they encountered an authentication issue because in a way it's advertises _"restricted data here - don't look!"_ 😅